.github/workflows/fuzz_test.yml

```diff
@@ -0,0 +1,156 @@
+    runs-on: ubuntu-latest
+    needs: fuzz-test-build-target
+    services:
+      minio:
+        image: wktk/minio-server
+        ports:
+          - 9000:9000
+        env:
+          MINIO_ACCESS_KEY: "minioadmin"
+          MINIO_SECRET_KEY: "minioadmin"
+    strategy:
+      fail-fast: true
+      matrix:
+        fuzz-targets: [ fuzz_reader, fuzz_range_reader, fuzz_writer ]
+    steps:
+      - name: Setup Test Bucket
+        env:
+          AWS_ACCESS_KEY_ID: "minioadmin"
+          AWS_SECRET_ACCESS_KEY: "minioadmin"
+          AWS_EC2_METADATA_DISABLED: "true"
+        run: aws --endpoint-url http://127.0.0.1:9000/ s3 mb s3://test
+      - name: Download Fuzz Targets
+        uses: actions/download-artifact@v3
+        with:
+          name: fuzz_targets
+          path: ./target
+      - name: Grant Execute Permissions
+        shell: bash
+        run: chmod +x ./target/${{ matrix.fuzz-targets }}
+      - name: Run Fuzz Test
+        shell: bash
+        run: ./target/${{ matrix.fuzz-targets }} -max_total_time=120
+        env:
+          OPENDAL_S3_TEST: on
+          OPENDAL_S3_BUCKET: test
+          OPENDAL_S3_ENDPOINT: "http://127.0.0.1:9000"
+          OPENDAL_S3_ACCESS_KEY_ID: minioadmin
+          OPENDAL_S3_SECRET_ACCESS_KEY: minioadmin
+      - name: Upload Crash Files
+        uses: actions/upload-artifact@v3
+        with:
+          name: fuzz_crashes
+          path: ./crash*
+  fuzz-test-run-fs:
+    runs-on: ubuntu-latest
+    needs: fuzz-test-build-target
+    strategy:
+      fail-fast: true
+      matrix:
+        fuzz-targets: [ fuzz_reader, fuzz_range_reader, fuzz_writer ]
+    steps:
+      - name: Download Fuzz Targets
+        uses: actions/download-artifact@v3
+        with:
+          name: fuzz_targets
+          path: ./target
+      - name: Grant Execute Permissions
+        shell: bash
+        run: chmod +x ./target/${{ matrix.fuzz-targets }}
+      - name: Run Fuzz Test
+        shell: bash
+        run: ./target/${{ matrix.fuzz-targets }} -max_total_time=120
+        env:
+          OPENDAL_FS_TEST: on
+          OPENDAL_FS_ROOT: ${{ runner.temp }}/
+      - name: Upload Crash Files
+        uses: actions/upload-artifact@v3
+        with:
+          name: fuzz_crashes
+          path: ./crash*
+  fuzz-test-run-memory:
+    runs-on: ubuntu-latest
+    needs: fuzz-test-build-target
+    strategy:
+      fail-fast: true
+      matrix:
+        fuzz-targets: [ fuzz_reader, fuzz_range_reader, fuzz_writer ]
+    steps:
+      - name: Download Fuzz Targets
+        uses: actions/download-artifact@v3
+        with:
+          name: fuzz_targets
+          path: ./target
+      - name: Grant Execute Permissions
+        shell: bash
+        run: chmod +x ./target/${{ matrix.fuzz-targets }}
+      - name: Run Fuzz Test
+        shell: bash
+        run: ./target/${{ matrix.fuzz-targets }} -max_total_time=120
+        env:
+          OPENDAL_MEMORY_TEST: on
+      - name: Upload Crash Files
```

Comment:
Failed test will skip this step, we need to add a if failure here.

Reply:
You are so familiar with that. Compile by brain 🤯